### PR TITLE
Renamed s/http2_/h2_

### PR DESCRIPTION
**Pull request by joedevivo**
@stillinbeta as promised.

### CODE REVIEW

File: src/h2_connection.erl [hunk 1]
```diff
@@ -558,8 +555,8 @@ route_frame({#frame_header{type=?HEADERS}=FH, _Payload}=Frame,
                       self(),
                       Conn#connection.stream_callback_mod,
                       Conn#connection.socket,
-                      Conn#connection.peer_settings#settings.initial_window_size,
-                      Conn#connection.self_settings#settings.initial_window_size,
+                      (Conn#connection.peer_settings)#settings.initial_window_size,
```
Comment (stillinbeta) [inline src/h2_connection.erl]: what's this for?

Comment (joedevivo) [inline src/h2_connection.erl]: The parenthesis? No idea. I guess wrangler did it, because I sure didn't.

File: test/client_server_SUITE.erl [hunk 2]
```diff
@@ -114,19 +114,19 @@ basic_push(_Config) ->
     %% We'll know we're done when we're notified of all the streams ending.
     wait_for_n_notifications(12),
 
-    Streams = http2_connection:get_streams(Client),
+    Streams = h2_connection:get_streams(Client),
     ct:pal("Streams ~p", [Streams]),
-    ?assertEqual(0, h2_stream_set:my_active_count(Streams)),
-    ?assertEqual(0, h2_stream_set:their_active_count(Streams)),
+    ?assertEqual(0, (h2_stream_set:my_active_count(Streams))),
```
Comment (stillinbeta) [inline test/client_server_SUITE.erl]: this seems kinda egregious

File: test/http2_spec_5_1_SUITE.erl [hunk 3]
```diff
@@ -212,53 +212,60 @@ half_closed_remote_sends_headers(_Config) ->
...
-    ?assertEqual(true, length(Resp) >= 3),
-    [ {HeadersH, _}, {DataH, _}, {RstStreamH, RstStream}|_] = Resp,
-    ?assertEqual(?HEADERS, HeadersH#frame_header.type),
-    ?assertEqual(?DATA, DataH#frame_header.type),
-    ?assertEqual(?RST_STREAM, RstStreamH#frame_header.type),
-    ?assertEqual(?STREAM_CLOSED, http2_frame_rst_stream:error_code(RstStream)),
+    ?assertEqual(true, (length(Resp) >= 1)),
+
+    %% We need to find if one of these things are a RstStream
+    RstStreams =
```
Comment (stillinbeta) [inline test/http2_spec_5_1_SUITE.erl]: trying to sneak some test changes in eh?

Comment (joedevivo) [inline test/http2_spec_5_1_SUITE.erl]: Nope! That's why I made it a separate commit *AND* told you "hey, the only thing really interesting here is this commit" :P

Comment (joedevivo) [inline test/http2_spec_5_1_SUITE.erl]: bottom line was that this test was only failing on Mac OS X, but consistently

**Comment by stillinbeta** [comment]
should we rename http2.hrl to h2.hrl?

**Comment by stillinbeta** [comment]
+1 from me with a few nits.

**Comment by joedevivo** [comment]
I think it's fine as `http2.hrl`. this was really about making module names more manageable in the code.

**Pull request merged**